### ISSUE TITLE
About the sample sequence length

### ISSUE THREAD

**Issue by gaozhiguang** [shwetanlp/CHQ-Summ#1]

**Comment by gaozhiguang** [comment]
Hi, very nive work.
function "train_batch_rl"  in main.py:
for di in range(max_decoding_step):
        outputs = model(encoder_outputs=encoder_outputs, decoder_input_ids=decoder_input_ids,
                        attention_mask=attention_mask,
                        use_cache=False
                        )
The parameter max_decoding_step is fixed, does it have a influence?
Thanks.

**Comment by gaozhiguang** [comment]
And what's more, 
if greedy is False:  # If multinomial based sampling, compute log probabilites of sampled words
        log_probs = torch.stack(log_probs, dim=1)
        log_probs = log_probs * target_mask  # Not considering sampled words with padding mask = 0
        lens = torch.sum(target_mask, dim=1)  # Length of sampled sentence
        log_probs = torch.sum(log_probs,
                              dim=1) / lens  # (bs,)   

the operation  log_probs = log_probs * target_mask works wrong, log_probs have a different size with target_mask.

**Comment by shwetanlp** [comment]
Hi, max_decoding_step is fixed in all the experiments, thus log_probs and target_mask are of the same size.
